chia/plot_sync/receiver.py

```diff
@@ -0,0 +1,313 @@
+import logging
+import time
+from typing import Any, Callable, Dict, List
+
+from chia.plot_sync.delta import Delta, PathListDelta, PlotListDelta
+from chia.plot_sync.exceptions import (
+    InvalidIdentifierError,
+    InvalidLastSyncIdError,
+    PlotAlreadyAvailableError,
+    PlotNotAvailableError,
+    PlotSyncException,
+    SyncIdsMatchError,
+)
+from chia.plot_sync.util import ErrorCodes, State
+from chia.protocols.harvester_protocol import (
+    Plot,
+    PlotSyncDone,
+    PlotSyncError,
+    PlotSyncIdentifier,
+    PlotSyncPathList,
+    PlotSyncPlotList,
+    PlotSyncResponse,
+    PlotSyncStart,
+)
+from chia.server.ws_connection import ProtocolMessageTypes, WSChiaConnection, make_msg
+from chia.util.ints import int16, uint64
+
+log = logging.getLogger(__name__)
+
+
+class Receiver:
+    _connection: WSChiaConnection
+    _sync_state: State
+    _delta: Delta
+    _expected_sync_id: uint64
+    _expected_message_id: uint64
+    _last_sync_id: uint64
+    _last_sync_time: float
+    _plots: Dict[str, Plot]
+    _invalid: List[str]
+    _keys_missing: List[str]
+    _duplicates: List[str]
+    _update_callback: Callable
+
+    def __init__(self, connection: WSChiaConnection, update_callback: Callable) -> None:
+        self._connection = connection
+        self._sync_state = State.idle
+        self._delta = Delta()
+        self._expected_sync_id = uint64(0)
+        self._expected_message_id = uint64(0)
+        self._last_sync_id = uint64(0)
+        self._last_sync_time = 0
+        self._plots = {}
+        self._invalid = []
+        self._keys_missing = []
+        self._duplicates = []
+        self._update_callback = update_callback  # type: ignore[assignment]
+
+    def reset(self) -> None:
+        self._sync_state = State.idle
+        self._expected_sync_id = uint64(0)
+        self._expected_message_id = uint64(0)
+        self._last_sync_id = uint64(0)
+        self._last_sync_time = 0
+        self._plots.clear()
+        self._invalid.clear()
+        self._keys_missing.clear()
+        self._duplicates.clear()
+        self._delta.clear()
+
+    def bump_expected_message_id(self) -> None:
+        self._expected_message_id = uint64(self._expected_message_id + 1)
+
+    def connection(self) -> WSChiaConnection:
```

Comment:
> Why all the privacy 

Because other classes shouldn't be able to just like this modify the internal state of receiver? Sure, its still possible but i imagine in a future time we want restrict access to private members in mypy?  

> and accessors?

To access them in tests without getting complaints from pycharm, i know im probably not consistent with this workflow in all places.. i.e. im accessing private stuff in other tests iirc :)